### ISSUE TITLE
[Issue]: Jellyfin won't read .cbr  also Bookshelf don't load metadata

### ISSUE THREAD

**Issue by Arturitu12** [jellyfin/jellyfin#7498]
### Please describe your bug

Hello, I'm trying to add my .cbr comic books to jellyfin but it won't load any content. Files works on local/online reader, but not in jellyfin. I created book-type library and added comics to them with name syntax form docs, I also have bookshelf plugin installed, but it doesn't create any metadata. I also see never ending sppining loading screen when I try to read anything. Please help! 

Steps to Reproduce:
1. Have book type library with .cbr files
2. Try to read anything from it
3. See never-ending loading loop

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Ubuntu 20.04 LTS
- Virtualization: none
- Clients: Any
- Browser: Any
- FFmpeg Version: -
- Playback Method: Direct Play
- Hardware Acceleration: none
- Plugins: Bookshelf
- Reverse Proxy: none
- Base URL: none
- Networking: NAT
- Storage: local
```

### Jellyfin logs

```shell
[2022-03-27 13:59:29.123 +02:00] [INF] Could not load ComicInfo metadata for "/biblioteka/komiksy/Batman Zero Year (2014)/Action Comics vol. 2 #25 (2014).cbr" from XML file
[2022-03-27 13:59:29.123 +02:00] [INF] Could not load ComicInfo metadata for "/biblioteka/komiksy/Batman Zero Year (2014)/Action Comics vol. 2 #25 (2014).cbr" from XML file. No internal XML in comic archive
[2022-03-27 13:59:29.313 +02:00] [INF] Could not load external xml from "/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml". This could mean there is no separate ComicInfo metadata file for this comic. Maybe the metadata is bundled within the comic
System.IO.FileNotFoundException: Could not find file '/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml'.
File name: '/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.Xml.XmlDownloadManager.<>c__DisplayClass1_0.<GetStreamAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at System.Xml.XmlUrlResolver.GetEntityAsync(Uri absoluteUri, String role, Type ofObjectToReturn)
   at System.Xml.XmlTextReaderImpl.FinishInitUriStringAsync()
   at System.Xml.AsyncHelper.CallBoolTaskFuncWhenFinishCoreAsync[TArg](Task task, Func`2 func, TArg arg)
   at System.Xml.Linq.XDocument.LoadAsyncInternal(XmlReader reader, LoadOptions options, CancellationToken cancellationToken)
   at Jellyfin.Plugin.Bookshelf.Providers.ComicBook.ExternalComicInfoProvider.LoadXml(ItemInfo info, IDirectoryService directoryService, CancellationToken cancellationToken)
[2022-03-27 13:59:29.317 +02:00] [INF] Could not load ComicInfo metadata for "/biblioteka/komiksy/Batman Zero Year (2014)/Red Hood and the Outlaws #25 (2014).cbr" from XML file
[2022-03-27 13:59:29.317 +02:00] [INF] Could not load ComicInfo metadata for "/biblioteka/komiksy/Batman Zero Year (2014)/Red Hood and the Outlaws #25 (2014).cbr" from XML file. No internal XML in comic archive
[2022-03-27 13:59:29.823 +02:00] [INF] Could not load external xml from "/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml". This could mean there is no separate ComicInfo metadata file for this comic. Maybe the metadata is bundled within the comic
System.IO.FileNotFoundException: Could not find file '/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml'.
File name: '/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.Xml.XmlDownloadManager.<>c__DisplayClass1_0.<GetStreamAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at System.Xml.XmlUrlResolver.GetEntityAsync(Uri absoluteUri, String role, Type ofObjectToReturn)
   at System.Xml.XmlTextReaderImpl.FinishInitUriStringAsync()
   at System.Xml.AsyncHelper.CallBoolTaskFuncWhenFinishCoreAsync[TArg](Task task, Func`2 func, TArg arg)
   at System.Xml.Linq.XDocument.LoadAsyncInternal(XmlReader reader, LoadOptions options, CancellationToken cancellationToken)
   at Jellyfin.Plugin.Bookshelf.Providers.ComicBook.ExternalComicInfoProvider.LoadXml(ItemInfo info, IDirectoryService directoryService, CancellationToken cancellationToken)
```

### Please attach any screenshots here

![image](https://user-images.githubusercontent.com/58103832/160280915-d0c4c8ac-f6e9-422b-99f2-2d274bd7de4d.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by MinecraftPlaye** [comment]
To fix your metadata:
`Could not load ComicInfo metadata for "/biblioteka/komiksy/Batman Zero Year (2014)/Red Hood and the Outlaws #25 (2014).cbr" from XML file`

This means that the bookshelf plug-in couldn't  load the metadata which might be placed inside the comic. This makes sense as the bookshelf plug-in can't open cbr files and check them for metadata. Only cbz files.

`Could not load external xml from "/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml".`
This means that the fallback did not work either as the bookshelf plugin could not find a file named `ComicInfo.xml` at the given path: `/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml`

The bookshelf plug-in itself doesn't (yet) grab any metadata from the internet except Google Books. It exclusively relies on local files providing the necessary information.


---

Regarding your issue with `cbr` files not loading: I can't help much as I don't have any of those. All I can say is, for cbz files, it might take a few minutes for the comic to load.

**Comment by Arturitu12** [comment]
> To fix your metadata: `Could not load ComicInfo metadata for "/biblioteka/komiksy/Batman Zero Year (2014)/Red Hood and the Outlaws #25 (2014).cbr" from XML file`
> 
> This means that the bookshelf plug-in couldn't load the metadata which might be placed inside the comic. This makes sense as the bookshelf plug-in can't open cbr files and check them for metadata. Only cbz files.
> 
> `Could not load external xml from "/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml".` This means that the fallback did not work either as the bookshelf plugin could not find a file named `ComicInfo.xml` at the given path: `/biblioteka/komiksy/Batman Zero Year (2014)/ComicInfo.xml` Placing a `ComicInfo.xml` file (which contains the metadata) at the given location should fix your metadata problem.
> 
> The bookshelf plug-in itself doesn't (yet) grab any metadata from the internet except from Google Books (in my experience, Google Books won't find many of my books and comics/mangas). It mainly relies on local files providing the necessary information.
> 
> Regarding your issue with `cbr` files not loading: I can't help much as I don't have any of those. All I can say is, for cbz files, it might take a few minutes for the comic to load.

I see about metadata, i'll add it manualy, but CBR won't start working! I tried with converted to Epub and all works fine but now i need to convert all my comics :(( Ok, i'll try but hope to see some better support in future. Cause wiki clearly claims that jellyfin reads CBR but it does not. Thanks anyway, i'll keep this isseu open as CBR support is still bugged

**Comment by Arturitu12** [comment]
I used ComicRack to convert my comics from CBR to CBZ with Comicinfo.xml it works now but i don't know why but in reader theres additional page at start:
![obraz_2022-03-28_174944](https://user-images.githubusercontent.com/58103832/160437241-fe8df3e0-13f2-46ed-a280-adbcd2a0b827.png)
Maybe reader also tries to read Comicinfo.xml ?

**Comment by MinecraftPlaye** [comment]
> Maybe reader also tries to read Comicinfo.xml ?

Yes. That's fixed in 10.8.

> but CBR won't start working!

Hm... not sure why. I don't have access right now to my 10.7 instance, but in my 10.8 development instance, cbr files did load on my PC. Thought I only tested it with a file I created out of two random images instead of a real comic.
Maybe someone else can help more with the cbr issue.

**Comment by craftersshaft** [comment]
is there a way to load cbr/cbz files without ComicInfo.xml? is there a program to generate it or should it show even without it?
